assets/src/edit-story/components/form/color.js

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { rgba } from 'polished';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+const Input = styled.input`
+  display: block;
+  appearance: none;
+  width: 32px;
+  height: 32px;
+  border: 1px solid ${({ theme }) => theme.colors.fg.v2} !important;
+  border-radius: 0 !important;
+  box-shadow: none !important;
+  outline: none;
+  padding: 0;
+  margin-right: ${({ suffix }) => (Boolean(suffix) ? 12 : 0)}px;
```

Comment:
Is this a universal `input` or is it meant to be specifically in the design panel? Because there are a lot of assumptions here for a very specific layout. Can these assumptions instead be moved to the container component?

Reply:
Just to clarify @dvoytenko: When you say to move this specificity to a container, are you saying to apply specificities only when we reuse the component somewhere as a child component? For instance:


```
<Container>
 <Button />
</Container>
```

or

```
const StyledButton = styled(Button)`...` 
```

Reply:
Yes. The default component would have no margin:

```
const Input = styled.input`
  ...
`;
```

And the call site, would customize it as needed for a specific situation:

```
import Input from './form/color';

const SpecialInput = styled(Input)`
  margin-right: 8px;
`;
```